urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/SevenZip/LzmaAlone.java

```diff
@@ -0,0 +1,331 @@
+package SevenZip;
+
+public class LzmaAlone
+{
+    static public class CommandLine
+    {
+        public static final int kEncode                 = 0;
+        public static final int kDecode                 = 1;
+        public static final int kBenchmak               = 2;
+        
+        public int              Command                 = -1;
+        public int              NumBenchmarkPasses      = 10;
+        
+        public int              DictionarySize          = 1 << 23;
+        public boolean          DictionarySizeIsDefined = false;
+        
+        public int              Lc                      = 3;
+        public int              Lp                      = 0;
+        public int              Pb                      = 2;
+        
+        public int              Fb                      = 128;
+        public boolean          FbIsDefined             = false;
+        
+        public boolean          Eos                     = false;
+        
+        public int              Algorithm               = 2;
+        public int              MatchFinder             = 1;
+        
+        public String           InFile;
+        public String           OutFile;
+        
+        boolean ParseSwitch(String s)
+        {
+            if (s.startsWith("d"))
+            {
+                DictionarySize = 1 << Integer.parseInt(s.substring(1));
+                DictionarySizeIsDefined = true;
+            }
+            else if (s.startsWith("fb"))
+            {
+                Fb = Integer.parseInt(s.substring(2));
+                FbIsDefined = true;
+            }
+            else if (s.startsWith("a"))
+            {
+                Algorithm = Integer.parseInt(s.substring(1));
+            }
+            else if (s.startsWith("lc"))
+            {
+                Lc = Integer.parseInt(s.substring(2));
+            }
+            else if (s.startsWith("lp"))
+            {
+                Lp = Integer.parseInt(s.substring(2));
+            }
+            else if (s.startsWith("pb"))
+            {
+                Pb = Integer.parseInt(s.substring(2));
+            }
+            else if (s.startsWith("eos"))
+            {
+                Eos = true;
+            }
+            else if (s.startsWith("mf"))
+            {
+                final String mfs = s.substring(2);
+                if (mfs.equals("bt2"))
+                {
+                    MatchFinder = 0;
+                }
+                else if (mfs.equals("bt4"))
+                {
+                    MatchFinder = 1;
+                }
+                else if (mfs.equals("bt4b"))
+                {
+                    MatchFinder = 2;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                return false;
+            }
+            return true;
+        }
+        
+        public boolean Parse(String[] args) throws Exception
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.